iterative/utils/analytics.go

```diff
@@ -0,0 +1,302 @@
+		os.Getenv("CI_PROJECT_ROOT_NAMESPACE"),
+		os.Getenv("BITBUCKET_WORKSPACE"),
+	))
+
+	if err != nil {
+		return ""
+	}
+
+	return id.String()
+}
+
+func UserId() string {
+	if IsCI() {
+		uid, err := deterministic(fmt.Sprintf("%s%s%s",
+			os.Getenv("GITHUB_ACTOR"),
+			os.Getenv("GITLAB_USER_ID"),
+			os.Getenv("BITBUCKET_STEP_TRIGGERER_UUID"),
+		))
+
+		if err == nil {
+			return uid.String()
+		}
+	}
+
+	id := uuid.New().String()
+	old := appdirs.UserConfigDir("dvc/user_id", "iterative", "", false)
+	new := appdirs.UserConfigDir("iterative/telemetry", "", "", false)
+
+	_, errorNew := os.Stat(new)
+
+	if os.IsNotExist(errorNew) {
+		_, errorOld := os.Stat(old)
+		if !os.IsNotExist(errorOld) {
+			jsonFile, jsonErr := os.Open(old)
+
+			if jsonErr == nil {
+				byteValue, _ := ioutil.ReadAll(jsonFile)
+				var data map[string]interface{}
+				json.Unmarshal([]byte(byteValue), &data)
+				id = data["user_id"].(string)
+
+				defer jsonFile.Close()
+			}
+		}
+
+		os.MkdirAll(filepath.Dir(new), 0644)
+		ioutil.WriteFile(new, []byte(id), 0644)
+	} else {
+		dat, _ := ioutil.ReadFile(new)
+		id = string(dat[:])
+	}
+
+	return id
+}
+
+func ResourceData(d *schema.ResourceData) map[string]interface{} {
+	if d == nil {
+		return map[string]interface{}{}
+	}
+
+	tpiLogs := d.Get("logs").([]interface{})
+	logs := ""
+	for _, log := range tpiLogs {
+		logs += log.(string)
+	}
+	spot := d.Get("spot").(float64)
+	return map[string]interface{}{
+		"cloud":           d.Get("cloud").(string),
+		"cloud_region":    d.Get("region").(string),
+		"cloud_machine":   d.Get("machine").(string),
+		"cloud_disk_size": d.Get("disk_size").(int),
+		"cloud_spot":      spot,
+		"cloud_spot_auto": spot == 0.0,
+		"task_status":     d.Get("status").(map[string]interface{}),
+		"task_duration":   TaskDuration(logs),
+		"task_resumed":    len(tpiLogs) > 1,
+	}
+}
+
+func JitsuEventPayload(action string, e error, extra map[string]interface{}) map[string]interface{} {
+	systemInfo := SystemInfo()
+
+	extra["ci"] = guessCI()
+	extra["terraform_version"] = TerraformVersion()
+
+	err := ""
+	if e != nil {
+		err = reflect.TypeOf(e).String()
+	}
+
+	payload := map[string]interface{}{
+		"user_id":      UserId(),
+		"group_id":     GroupId(),
+		"action":       action,
+		"interface":    "cli",
+		"tool_name":    "tpi",
+		"tool_source":  "terraform",
+		"tool_version": Version,
+		"os_name":      systemInfo["platform"],
+		"os_version":   systemInfo["platform_version"],
+		"backend":      extra["cloud"],
+		"error":        err,
+		"extra":        extra,
+	}
+
+	return payload
+}
+
+func SendJitsuEvent(action string, e error, extra map[string]interface{}) {
+	for _, env := range []string{"DVC_NO_ANALYTICS"} {
+		if _, ok := os.LookupEnv(env); ok {
+			logrus.Debugf("analytics: %s environment variable is set; doing nothing", env)
+			return
+		}
+	}
+
+	go send(JitsuEventPayload(action, e, extra))
+	wg.Add(1)
+}
+
+func send(event interface{}) error {
+	defer wg.Done()
+
```

Comment:
should check `if event.user_id == "do-not-track" { return nil }`

Reply:
Its handled in 

```go
func SendJitsuEvent(action string, e error, extra map[string]interface{}) {
	for _, env := range []string{"ITERATIVE_DO_NOT_TRACK"} {
		if _, ok := os.LookupEnv(env); ok {
			logrus.Debugf("analytics: %s environment variable is set; doing nothing", env)
			return
		}
	}

	go send(JitsuEventPayload(action, e, extra))
	wg.Add(1)
}
```